### ISSUE TITLE
OfficeCaltech results

### ISSUE THREAD

**Issue by CarlaMalo** [jyzgh/DapperFL#2]
Hello,
We are trying to reproduce the results from the paper for the  fl_officecaltech dataset using the configuration from the Appendix B, Table 4: Default hyper-parameters. But we are getting really different results, specially for the Webcam domain we observed half of the accuracy (instead of 74.14%, we got 39.66%). Is there anything else we might need to consider?

* This is the dataset source: https://github.com/ZhangJUJU/OfficeCaltechDomainAdaptation/tree/master
* Parameter configuration:
reg_coeff = 0.01
wandb = 1
device_id = 0
rand_dataset = 1
learning_decay = False
averaging = 'weight'
online_ratio = 1
csv_log = False
local_lr = 0.01
local_batch_size = 64
pr_ratios = ['0', '0.2', '0.4', '0.6', '0.8']
arch = 'res18'
local_epoch = 5
parti_num =10
model = 'dapperfl'
dataset = 'fl_officecaltech'
pr_strategy = 'AD'
backbone = 'resnet18'
alpha = 0.9
alpha_min = 0.1
epsilon = 0.2

I've attached the results. 

![Image](https://github.com/user-attachments/assets/1e20e8a7-f311-44e3-9807-a99cab4e69ec)

Thanks for the support!

**Comment by YongzheJia** [comment]
Hi, CarlaMalo:

From the picture and your description, it looks like the issue might be caused by too few clients being assigned to the Webcam domain, or by a biased selection of data from that domain.

In our setup, only 20% of the Webcam training data is used for local training in DapperFL. So, if there are too few clients assigned to Webcam, the model might not capture the full data distribution well — which could lead to unstable or lower accuracy.

To improve this, you could try a few things: 1) check how many clients are assigned to the Webcam domain and ensure that it’s sufficient (e.g., more than 2 clients), 2) try different random seeds to re-sample the data and compute the average results, 3) increase the sampling ratio for the dataset to get more stable results.

Hope this helps!
